utility_analysis_new/combiners.py

```diff
@@ -552,22 +574,26 @@ def __init__(self, params: pipeline_dp.combiners.CombinerParams,
         self._params = params
         self._error_quantiles = error_quantiles
 
-    def create_accumulator(
-            self, metrics: CountUtilityAnalysisMetrics) -> AccumulatorType:
+    def create_accumulator(self,
+                           metrics: CountUtilityAnalysisMetrics,
+                           probability_to_keep: float = 1) -> AccumulatorType:
         """Creates an accumulator for metrics."""
         # Absolute error metrics
-        abs_error_expected = metrics.per_partition_error + metrics.expected_cross_partition_error
-        abs_error_variance = metrics.std_cross_partition_error**2 + metrics.std_noise**2
+        abs_error_expected = probability_to_keep * (
```

Comment:
We need to compute weighted average. The weighted average of `a1,..,an` with weight `w1,..,wn` equals to
`(w1a1 + ... + wnan)/(w1+...+ wn)`.
Here weights are `probability_to_keep`, so we need to compute sum of weights and divide by sum of weight, not on `num_partitions` (this sum is equal to num_partition when all `probability_to_keep=1`)